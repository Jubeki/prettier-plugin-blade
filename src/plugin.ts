import { AstPath, format } from "prettier";
import { Lexer, Parser } from "./lang";
import { Plugin, Doc } from "prettier";