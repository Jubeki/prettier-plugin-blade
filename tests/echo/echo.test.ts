import fs from "fs";
import path from "path";
import prettier from "prettier";
import plugin from "../../src/plugin";

const fixture = fs.readFileSync(path.join(__dirname, "./echo.blade.php"), "utf-8");

function format(content: string) {
    return prettier.format(content, {
        parser: "blade",
        plugins: [plugin as any as string]
    });
}

test("basic", () => {
    const formatted = format(fixture);
    expect(formatted).toMatchSnapshot();
});
