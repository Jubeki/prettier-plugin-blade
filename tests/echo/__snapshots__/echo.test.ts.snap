// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"<h1>
  Here's some very poorly formatted {{ \\"html\\" }} with weird {{ $echoParts + 1 / $insideOfIt }}
</h1>
"
`;
